base_layer/core/src/test_helpers/blockchain.rs

```diff
@@ -45,10 +73,199 @@ pub fn create_new_blockchain() -> BlockchainDatabase<MemoryDatabase<HashDigest>>
         })
         .on_ties(ChainStrengthComparerBuilder::new().by_height().build())
         .build();
-    super::create_mem_db(&consensus_manager)
+    let validators = Validators::new(MockValidator::new(true), MockValidator::new(true));
+    create_store_with_consensus_and_validators(&consensus_manager, validators)
 }
 
 fn genesis_template() -> NewBlockTemplate {
     let header = BlockHeader::new(0);
     header.into_builder().build().into()
 }
+
+pub fn create_store_with_consensus_and_validators(
+    rules: &ConsensusManager,
+    validators: Validators<TempDatabase>,
+) -> BlockchainDatabase<TempDatabase>
+{
+    let backend = create_test_db();
+    BlockchainDatabase::new(backend, &rules, validators, BlockchainDatabaseConfig::default(), false).unwrap()
+}
+
+pub fn create_store_with_consensus(rules: &ConsensusManager) -> BlockchainDatabase<TempDatabase> {
+    let factories = CryptoFactories::default();
+    let validators = Validators::new(
+        FullConsensusValidator::new(rules.clone()),
+        StatelessBlockValidator::new(rules.clone(), factories),
+    );
+    create_store_with_consensus_and_validators(rules, validators)
+}
+pub fn create_store() -> BlockchainDatabase<TempDatabase> {
+    let network = Network::Ridcully;
+    let rules = ConsensusManagerBuilder::new(network).build();
+    create_store_with_consensus(&rules)
+}
+
+pub fn create_test_db() -> TempDatabase {
+    TempDatabase::new()
+}
+
+pub struct TempDatabase {
+    path: PathBuf,
+    db: LMDBDatabase,
+}
+
+impl TempDatabase {
+    fn new() -> Self {
+        let temp_path = create_temporary_data_path();
+
+        Self {
+            db: create_lmdb_database(&temp_path, LMDBConfig::default()).unwrap(),
+            path: temp_path,
+        }
+    }
+}
+
+impl Deref for TempDatabase {
+    type Target = LMDBDatabase;
+
+    fn deref(&self) -> &Self::Target {
+        &self.db
+    }
+}
+
+impl Drop for TempDatabase {
+    fn drop(&mut self) {
+        if std::path::Path::new(&self.path).exists() {
+            match std::fs::remove_dir_all(&self.path) {
+                Err(e) => println!("\n{:?}\n", e),
+                _ => (),
+            }
+        }
+    }
+}
+
```

Comment:
Could you `impl Borrow<BlockchainBackend>` or `AsRef` here to remove all this boilerplate?